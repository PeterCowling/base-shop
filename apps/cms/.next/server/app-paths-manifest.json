{
  "/_not-found/page": "app/_not-found/page.js",
  "/favicon.ico/route": "app/favicon.ico/route.js",
  "/(cms)/products/page": "app/(cms)/products/page.js",
  "/(cms)/products/[id]/edit/page": "app/(cms)/products/[id]/edit/page.js",
  "/[lang]/page": "app/[lang]/page.js"
}