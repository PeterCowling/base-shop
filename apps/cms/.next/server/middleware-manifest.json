{
  "version": 3,
  "middleware": {
    "/": {
      "files": [
        "server/edge-runtime-webpack.js",
        "server/src/middleware.js"
      ],
      "name": "src/middleware",
      "page": "/",
      "matchers": [
        {
          "regexp": "^(?:\\/(_next\\/data\\/[^/]{1,}))?(?:\\/((?!api|_next|.*\\.[\\w]+$).*))(\\.json)?[\\/#\\?]?$",
          "originalSource": "/((?!api|_next|.*\\.[\\w]+$).*)"
        }
      ],
      "wasm": [],
      "assets": [],
      "env": {
        "__NEXT_BUILD_ID": "R97IXOVHNL_gddSM9cb2E",
        "NEXT_SERVER_ACTIONS_ENCRYPTION_KEY": "eLDaZ9079Hhu6ineRTE4EeGNmhHI6djndNvYUKNa80M=",
        "__NEXT_PREVIEW_MODE_ID": "821ac31694e6f092d06d52fe0910f46f",
        "__NEXT_PREVIEW_MODE_ENCRYPTION_KEY": "02b3c86f2fbc3b64e871693434c419793179868748674b1d138c80d026eff20d",
        "__NEXT_PREVIEW_MODE_SIGNING_KEY": "350bca960070f4777dc082ccd8947cb2e4ab4ecec1c235f90b09980f255a76dd"
      }
    }
  },
  "functions": {
    "/api/cart/route": {
      "files": [
        "server/server-reference-manifest.js",
        "server/app/api/cart/route_client-reference-manifest.js",
        "server/middleware-build-manifest.js",
        "server/middleware-react-loadable-manifest.js",
        "server/next-font-manifest.js",
        "server/interception-route-rewrite-manifest.js",
        "server/edge-runtime-webpack.js",
        "server/edge-chunks/935.js",
        "server/app/api/cart/route.js"
      ],
      "name": "app/api/cart/route",
      "page": "/api/cart/route",
      "matchers": [
        {
          "regexp": "^/api/cart$",
          "originalSource": "/api/cart"
        }
      ],
      "wasm": [],
      "assets": [],
      "env": {
        "__NEXT_BUILD_ID": "R97IXOVHNL_gddSM9cb2E",
        "NEXT_SERVER_ACTIONS_ENCRYPTION_KEY": "eLDaZ9079Hhu6ineRTE4EeGNmhHI6djndNvYUKNa80M=",
        "__NEXT_PREVIEW_MODE_ID": "821ac31694e6f092d06d52fe0910f46f",
        "__NEXT_PREVIEW_MODE_ENCRYPTION_KEY": "02b3c86f2fbc3b64e871693434c419793179868748674b1d138c80d026eff20d",
        "__NEXT_PREVIEW_MODE_SIGNING_KEY": "350bca960070f4777dc082ccd8947cb2e4ab4ecec1c235f90b09980f255a76dd"
      }
    },
    "/api/checkout-session/route": {
      "files": [
        "server/server-reference-manifest.js",
        "server/app/api/checkout-session/route_client-reference-manifest.js",
        "server/middleware-build-manifest.js",
        "server/middleware-react-loadable-manifest.js",
        "server/next-font-manifest.js",
        "server/interception-route-rewrite-manifest.js",
        "server/edge-runtime-webpack.js",
        "server/edge-chunks/935.js",
        "server/app/api/checkout-session/route.js"
      ],
      "name": "app/api/checkout-session/route",
      "page": "/api/checkout-session/route",
      "matchers": [
        {
          "regexp": "^/api/checkout\\-session$",
          "originalSource": "/api/checkout-session"
        }
      ],
      "wasm": [],
      "assets": [],
      "env": {
        "__NEXT_BUILD_ID": "R97IXOVHNL_gddSM9cb2E",
        "NEXT_SERVER_ACTIONS_ENCRYPTION_KEY": "eLDaZ9079Hhu6ineRTE4EeGNmhHI6djndNvYUKNa80M=",
        "__NEXT_PREVIEW_MODE_ID": "821ac31694e6f092d06d52fe0910f46f",
        "__NEXT_PREVIEW_MODE_ENCRYPTION_KEY": "02b3c86f2fbc3b64e871693434c419793179868748674b1d138c80d026eff20d",
        "__NEXT_PREVIEW_MODE_SIGNING_KEY": "350bca960070f4777dc082ccd8947cb2e4ab4ecec1c235f90b09980f255a76dd"
      }
    }
  },
  "sortedMiddleware": [
    "/"
  ]
}