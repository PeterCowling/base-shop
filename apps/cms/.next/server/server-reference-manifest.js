self.__RSC_SERVER_MANIFEST="{\n  \"node\": {\n    \"60ca58e256aff7a649cfcf77e7659d682ee44d86e8\": {\n      \"workers\": {\n        \"app/(cms)/shop/[shop]/products/page\": {\n          \"moduleId\": \"(rsc)/../../node_modules/.pnpm/next@15.3.4_@babel+core@7.27.4_@playwright+test@1.53.1_react-dom@19.1.0_react@19.1.0__react@19.1.0/node_modules/next/dist/build/webpack/loaders/next-flight-action-entry-loader.js?actions=%5B%5B%22%2FUsers%2Fpetercowling%2Freception%2Fbase-shop%2Fapps%2Fcms%2Fsrc%2Fapp%2F(cms)%2Fshop%2F%5Bshop%5D%2Fproducts%2Fpage.tsx%22%2C%5B%7B%22id%22%3A%2260ca58e256aff7a649cfcf77e7659d682ee44d86e8%22%2C%22exportedName%22%3A%22%24%24RSC_SERVER_ACTION_0%22%7D%5D%5D%2C%5B%22%2FUsers%2Fpetercowling%2Freception%2Fbase-shop%2Fapps%2Fcms%2Fsrc%2Factions%2Fproducts.ts%22%2C%5B%7B%22id%22%3A%22401e3b1c76c6824d4cde4384dbbf4852d746bbed32%22%2C%22exportedName%22%3A%22createDraft%22%7D%2C%7B%22id%22%3A%2240775ea723f47528880f853d1f21b19a4af756f97d%22%2C%22exportedName%22%3A%22createDraftRecord%22%7D%2C%7B%22id%22%3A%22603631e271d7d5e01f97d89a7edc5044e27f828180%22%2C%22exportedName%22%3A%22updateProduct%22%7D%5D%5D%5D&__client_imported__=!\",\n          \"async\": false\n        }\n      },\n      \"layer\": {\n        \"app/(cms)/shop/[shop]/products/page\": \"rsc\"\n      }\n    },\n    \"401e3b1c76c6824d4cde4384dbbf4852d746bbed32\": {\n      \"workers\": {\n        \"app/(cms)/shop/[shop]/products/page\": {\n          \"moduleId\": \"(rsc)/../../node_modules/.pnpm/next@15.3.4_@babel+core@7.27.4_@playwright+test@1.53.1_react-dom@19.1.0_react@19.1.0__react@19.1.0/node_modules/next/dist/build/webpack/loaders/next-flight-action-entry-loader.js?actions=%5B%5B%22%2FUsers%2Fpetercowling%2Freception%2Fbase-shop%2Fapps%2Fcms%2Fsrc%2Fapp%2F(cms)%2Fshop%2F%5Bshop%5D%2Fproducts%2Fpage.tsx%22%2C%5B%7B%22id%22%3A%2260ca58e256aff7a649cfcf77e7659d682ee44d86e8%22%2C%22exportedName%22%3A%22%24%24RSC_SERVER_ACTION_0%22%7D%5D%5D%2C%5B%22%2FUsers%2Fpetercowling%2Freception%2Fbase-shop%2Fapps%2Fcms%2Fsrc%2Factions%2Fproducts.ts%22%2C%5B%7B%22id%22%3A%22401e3b1c76c6824d4cde4384dbbf4852d746bbed32%22%2C%22exportedName%22%3A%22createDraft%22%7D%2C%7B%22id%22%3A%2240775ea723f47528880f853d1f21b19a4af756f97d%22%2C%22exportedName%22%3A%22createDraftRecord%22%7D%2C%7B%22id%22%3A%22603631e271d7d5e01f97d89a7edc5044e27f828180%22%2C%22exportedName%22%3A%22updateProduct%22%7D%5D%5D%5D&__client_imported__=!\",\n          \"async\": false\n        }\n      },\n      \"layer\": {\n        \"app/(cms)/shop/[shop]/products/page\": \"rsc\"\n      }\n    },\n    \"40775ea723f47528880f853d1f21b19a4af756f97d\": {\n      \"workers\": {\n        \"app/(cms)/shop/[shop]/products/page\": {\n          \"moduleId\": \"(rsc)/../../node_modules/.pnpm/next@15.3.4_@babel+core@7.27.4_@playwright+test@1.53.1_react-dom@19.1.0_react@19.1.0__react@19.1.0/node_modules/next/dist/build/webpack/loaders/next-flight-action-entry-loader.js?actions=%5B%5B%22%2FUsers%2Fpetercowling%2Freception%2Fbase-shop%2Fapps%2Fcms%2Fsrc%2Fapp%2F(cms)%2Fshop%2F%5Bshop%5D%2Fproducts%2Fpage.tsx%22%2C%5B%7B%22id%22%3A%2260ca58e256aff7a649cfcf77e7659d682ee44d86e8%22%2C%22exportedName%22%3A%22%24%24RSC_SERVER_ACTION_0%22%7D%5D%5D%2C%5B%22%2FUsers%2Fpetercowling%2Freception%2Fbase-shop%2Fapps%2Fcms%2Fsrc%2Factions%2Fproducts.ts%22%2C%5B%7B%22id%22%3A%22401e3b1c76c6824d4cde4384dbbf4852d746bbed32%22%2C%22exportedName%22%3A%22createDraft%22%7D%2C%7B%22id%22%3A%2240775ea723f47528880f853d1f21b19a4af756f97d%22%2C%22exportedName%22%3A%22createDraftRecord%22%7D%2C%7B%22id%22%3A%22603631e271d7d5e01f97d89a7edc5044e27f828180%22%2C%22exportedName%22%3A%22updateProduct%22%7D%5D%5D%5D&__client_imported__=!\",\n          \"async\": false\n        }\n      },\n      \"layer\": {\n        \"app/(cms)/shop/[shop]/products/page\": \"rsc\"\n      }\n    },\n    \"603631e271d7d5e01f97d89a7edc5044e27f828180\": {\n      \"workers\": {\n        \"app/(cms)/shop/[shop]/products/page\": {\n          \"moduleId\": \"(rsc)/../../node_modules/.pnpm/next@15.3.4_@babel+core@7.27.4_@playwright+test@1.53.1_react-dom@19.1.0_react@19.1.0__react@19.1.0/node_modules/next/dist/build/webpack/loaders/next-flight-action-entry-loader.js?actions=%5B%5B%22%2FUsers%2Fpetercowling%2Freception%2Fbase-shop%2Fapps%2Fcms%2Fsrc%2Fapp%2F(cms)%2Fshop%2F%5Bshop%5D%2Fproducts%2Fpage.tsx%22%2C%5B%7B%22id%22%3A%2260ca58e256aff7a649cfcf77e7659d682ee44d86e8%22%2C%22exportedName%22%3A%22%24%24RSC_SERVER_ACTION_0%22%7D%5D%5D%2C%5B%22%2FUsers%2Fpetercowling%2Freception%2Fbase-shop%2Fapps%2Fcms%2Fsrc%2Factions%2Fproducts.ts%22%2C%5B%7B%22id%22%3A%22401e3b1c76c6824d4cde4384dbbf4852d746bbed32%22%2C%22exportedName%22%3A%22createDraft%22%7D%2C%7B%22id%22%3A%2240775ea723f47528880f853d1f21b19a4af756f97d%22%2C%22exportedName%22%3A%22createDraftRecord%22%7D%2C%7B%22id%22%3A%22603631e271d7d5e01f97d89a7edc5044e27f828180%22%2C%22exportedName%22%3A%22updateProduct%22%7D%5D%5D%5D&__client_imported__=!\",\n          \"async\": false\n        }\n      },\n      \"layer\": {\n        \"app/(cms)/shop/[shop]/products/page\": \"rsc\"\n      }\n    }\n  },\n  \"edge\": {},\n  \"encryptionKey\": \"process.env.NEXT_SERVER_ACTIONS_ENCRYPTION_KEY\"\n}"