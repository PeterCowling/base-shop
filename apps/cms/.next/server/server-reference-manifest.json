{
  "node": {
    "001e3b1c76c6824d4cde4384dbbf4852d746bbed32": {
      "workers": {
        "app/(cms)/products/page": {
          "moduleId": "(rsc)/../../node_modules/.pnpm/next@15.3.4_@babel+core@7.27.4_@playwright+test@1.53.1_react-dom@19.1.0_react@19.1.0__react@19.1.0/node_modules/next/dist/build/webpack/loaders/next-flight-action-entry-loader.js?actions=%5B%5B%22%2FUsers%2Fpetercowling%2Freception%2Fbase-shop%2Fapps%2Fcms%2Fsrc%2Factions%2Fproducts.ts%22%2C%5B%7B%22id%22%3A%22001e3b1c76c6824d4cde4384dbbf4852d746bbed32%22%2C%22exportedName%22%3A%22createDraft%22%7D%2C%7B%22id%22%3A%2200775ea723f47528880f853d1f21b19a4af756f97d%22%2C%22exportedName%22%3A%22createDraftRecord%22%7D%5D%5D%5D&__client_imported__=!",
          "async": false
        }
      },
      "layer": {
        "app/(cms)/products/page": "rsc"
      }
    },
    "00775ea723f47528880f853d1f21b19a4af756f97d": {
      "workers": {
        "app/(cms)/products/page": {
          "moduleId": "(rsc)/../../node_modules/.pnpm/next@15.3.4_@babel+core@7.27.4_@playwright+test@1.53.1_react-dom@19.1.0_react@19.1.0__react@19.1.0/node_modules/next/dist/build/webpack/loaders/next-flight-action-entry-loader.js?actions=%5B%5B%22%2FUsers%2Fpetercowling%2Freception%2Fbase-shop%2Fapps%2Fcms%2Fsrc%2Factions%2Fproducts.ts%22%2C%5B%7B%22id%22%3A%22001e3b1c76c6824d4cde4384dbbf4852d746bbed32%22%2C%22exportedName%22%3A%22createDraft%22%7D%2C%7B%22id%22%3A%2200775ea723f47528880f853d1f21b19a4af756f97d%22%2C%22exportedName%22%3A%22createDraftRecord%22%7D%5D%5D%5D&__client_imported__=!",
          "async": false
        }
      },
      "layer": {
        "app/(cms)/products/page": "rsc"
      }
    },
    "40e5565d736feb51280e83f995fcd8a82991f0b795": {
      "workers": {
        "app/(cms)/products/[id]/edit/page": {
          "moduleId": "(action-browser)/../../node_modules/.pnpm/next@15.3.4_@babel+core@7.27.4_@playwright+test@1.53.1_react-dom@19.1.0_react@19.1.0__react@19.1.0/node_modules/next/dist/build/webpack/loaders/next-flight-action-entry-loader.js?actions=%5B%5B%22%2FUsers%2Fpetercowling%2Freception%2Fbase-shop%2Fapps%2Fcms%2Fsrc%2Factions%2FupdateProduct.ts%22%2C%5B%7B%22id%22%3A%2240e5565d736feb51280e83f995fcd8a82991f0b795%22%2C%22exportedName%22%3A%22updateProduct%22%7D%5D%5D%5D&__client_imported__=true!",
          "async": false
        }
      },
      "layer": {
        "app/(cms)/products/[id]/edit/page": "action-browser"
      }
    }
  },
  "edge": {},
  "encryptionKey": "eLDaZ9079Hhu6ineRTE4EeGNmhHI6djndNvYUKNa80M="
}