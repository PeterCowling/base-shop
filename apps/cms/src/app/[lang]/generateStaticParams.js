"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = generateStaticParams;
// apps/cms/src/app/[lang]/generateStaticParams.ts
var i18n_1 = require("@acme/i18n");
function generateStaticParams() {
    return i18n_1.LOCALES.map(function (lang) { return ({ lang: lang }); });
}
