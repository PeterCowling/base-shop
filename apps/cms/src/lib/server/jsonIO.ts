import "server-only";

import { promises as fs } from "fs";
import * as path from "path";

const fileLocks = new Map<string, Promise<void>>();

async function withFileLock<T>(file: string, fn: () => Promise<T>): Promise<T> {
  const prev = fileLocks.get(file) ?? Promise.resolve();
  let release: (() => void) | undefined;
  const current = new Promise<void>((resolve) => {
    release = resolve;
  });
  const next = prev.then(() => current);
  fileLocks.set(file, next);
  await prev;
  try {
    return await fn();
  } finally {
    release?.();
    if (fileLocks.get(file) === next) {
      fileLocks.delete(file);
    }
  }
}

export async function readJsonFile<T>(file: string, fallback: T): Promise<T> {
  try {
    // Callers pass validated paths rooted in the workspace data directory
    // eslint-disable-next-line security/detect-non-literal-fs-filename
    const buf = await fs.readFile(file, "utf8");
    return JSON.parse(buf) as T;
  } catch {
    return fallback;
  }
}

export async function writeJsonFile(
  file: string,
  value: unknown,
  indent?: number,
): Promise<void> {
  if (value === undefined || value === null || typeof value !== "object") {
    throw new TypeError("Expected value to be a non-null object or array");
  }

  // eslint-disable-next-line security/detect-non-literal-fs-filename
  await fs.mkdir(path.dirname(file), { recursive: true });
  const data = JSON.stringify(value, null, indent ?? 2);

  if (typeof fs.rename === "function") {
    const tmp = `${file}.${process.pid}.${Date.now()}.tmp`;
    // eslint-disable-next-line security/detect-non-literal-fs-filename
    await fs.writeFile(tmp, data, "utf8");
    // eslint-disable-next-line security/detect-non-literal-fs-filename
    await fs.rename(tmp, file);
  } else {
    // eslint-disable-next-line security/detect-non-literal-fs-filename
    await fs.writeFile(file, data, "utf8");
  }
}

export { withFileLock };
