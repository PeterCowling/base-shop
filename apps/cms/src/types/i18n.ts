import enJson from "@i18n/en.json";

export type Messages = typeof enJson;
