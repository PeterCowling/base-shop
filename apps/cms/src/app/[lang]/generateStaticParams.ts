// apps/cms/src/app/[lang]/generateStaticParams.ts
import { LOCALES } from "@acme/i18n";

export default function generateStaticParams() {
  return LOCALES.map((lang) => ({ lang }));
}
