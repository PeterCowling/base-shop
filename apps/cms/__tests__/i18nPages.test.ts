// apps/cms/__tests__/i18nPages.test.ts
// -------------------------------------------------------
// This test file is now a proper ESM / JSX module.
// -------------------------------------------------------

import React from "react";
import { renderToStaticMarkup } from "react-dom/server";

// Static imports because Next 12-15 can handle literal “[lang]” folders.
import { LOCALES } from "@acme/i18n";
import generateStaticParams from "../src/app/[lang]/generateStaticParams";
import LocaleLayout from "../src/app/[lang]/layout";

describe("i18n helpers", () => {
  it("generateStaticParams yields locales", () => {
    const params = generateStaticParams();

    for (const locale of LOCALES) {
      expect(params).toContainEqual({ lang: locale });
    }

    expect(params).toMatchSnapshot();
  });

  it("layout wraps children with TranslationsProvider", async () => {
    const html = renderToStaticMarkup(
      await LocaleLayout({
        children: React.createElement("div", { id: "c" }),
        // Next 15 delivers a *Promise*, so emulate that here.
        params: Promise.resolve({ lang: "de" }),
      })
    );

    expect(html).toContain('id="c"');
  });
});
