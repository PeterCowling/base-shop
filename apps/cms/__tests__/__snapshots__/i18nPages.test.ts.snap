// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18n helpers generateStaticParams yields locales 1`] = `
[
  {
    "lang": "en",
  },
  {
    "lang": "de",
  },
  {
    "lang": "it",
  },
]
`;